cmd/service.go

```diff
@@ -28,45 +32,67 @@ var serviceCmd = &cobra.Command{
 }
 
 var serviceCreateCmd = &cobra.Command{
-	Use:   "create <service_type> [service_name]",
+	Use:   "create <service_type> --plan <plan_name> [service_name]",
 	Short: "Create a new service",
-	Long: `Create a new service from service catalog to deploy on OpenShift.
+	Long: `Create a new service from service catalog using the plan defined and deploy it on OpenShift.
 
-If service name is not provided, service type value will be used.
+If service name is not provided, service type value will be used. The plan to be used must be passed along the service type
+using this convention <service_type>/<plan>. The parameters to configure the service are passed as a list of key=value pairs.
+The list of the parameters and their type is defined according to the plan selected.
 
 A full list of service types that can be deployed are available using: 'odo catalog list services'`,
-	Example: `  # Create new mysql-persistent service from service catalog.
-  odo service create mysql-persistent
+	Example: `  # Create new postgresql service from service catalog using dev plan and name my-postgresql-db.
+  odo service create dh-postgresql-apb my-postgresql-db --plan dev -p postgresql_user=luke -p postgresql_password=secret
```

Comment:
I'd rather see explicit flags than relying on argument positions: `--class dh-postgresql-apb --instance my-postgresql-db`. Another thing I don't like is mixing long and short flags. In this case, `--plan` doesn't have a short version because it would conflict with `-p` for the parameters. Could we find a way to have both long and short flags for all flags?

Reply:
That's exactly why I didn't add a short flag for plan. What do you suggest?

Relying on positional arguments although perhaps debatable, is the way `odo` works currently from most commands I believe

Reply:
I haven't reviewed all commands yet but UX-wise, positional arguments are a bad idea as they are really error-prone and add to the user's cognitive load for no good reason.

Reply:
Consistency is important. I'm also not a big fan of how we can specify a different name using positional argument. I believe that positional arguments should be used only if the value is required.

But right now I would rather go with how we do it in other places and have it at in a consistent way.
Let's go with positional arguments in the same way it is done for components storages and urls.

We can look into this later, design it in a better way and change it everywhere.

Reply:
Thanks for the input @kadel